src/Drivers/Saman/Saman.php

```diff
@@ -52,13 +52,18 @@ public function purchase()
             'MID' => $this->settings->merchantId,
             'ResNum' => $this->invoice->getUuid(),
             'Amount' => $this->invoice->getAmount() * 10, // convert to rial
+            'AdditionalData1' => $this->invoice->getDetails()[0], // used for getting reports, due to https://www.sep.ir/Data/Sites/1/media/token98.pdf
+            'AdditionalData2' => $this->invoice->getDetails()[1], // used for getting reports, due to https://www.sep.ir/Data/Sites/1/media/token98.pdf
+            'wage' => 0, // TODO, wage that should be payed by customer 
```

Comment:
it is TODO ? i cant merge TODO codes! it must be tested and ready to production.
and what is wage? can you tell me how can we use it?

Reply:
maybe it was better that i made an issue. excuse me.

wage is the amount of money can be set by seller to be costed from customer. for example a percentage of tax.
your invoice class doesn't have the wage field and it's 0 for all drivers. at the moment its zero too. 
i just wanted to inform you about that.

and maybe another TODO, is that AdditionalData1 and 2 can accept only one value. more information about their allowed types is not provided in docs, https://www.sep.ir/Data/Sites/1/media/token98.pdf.

so if it cant accept arrays or something, we should throw an exception. 

i just wanted to show you the issue. and it was better to create an issue instead.  

@khanzadimahdi

Reply:
taxed must be handled before payment request! you can calculate and add some taxed into payment price and then purchase the final price! I think we don't need to delegate tax calculation to bank gateway!

thanks Okay! so **create and issue and close this pull request** please.

thanks for your contribution.

Reply:
OK, NP.
Thanks.